src/VisualStudio/CSharp/Impl/ProjectSystemShim/CSharpProjectShim.cs

```diff
@@ -92,8 +87,12 @@ private string GetIdForErrorCode(int errorCode)
             return "CS" + errorCode.ToString("0000");
         }
 
-        protected CSharpCompilationOptions CreateCompilationOptions()
+        protected override CompilationOptions GetCompilationOptions(ParseOptions newParseOptions)
         {
+            // Get the base options from command line arguments + common workspace defaults.
+            var options = (CSharpCompilationOptions)base.GetCompilationOptions(newParseOptions);
```

Comment:
You could make this type paramtereized on the type of the compilation options it has.  then you won't need these types of casts in the code.